demo/app/lib/views/otp.dart

```diff
@@ -150,28 +185,36 @@ class _OTPPage extends State<OTP> {
                           _navigateToCredPreviewScreen(credentials, issuerURI, serializeDisplayData!, userDIDId, credID);
                         } catch (err) {
                           String errorMessage = err.toString();
+                          log("errorMessage-> $errorMessage");
                           if (err is PlatformException &&
                               err.message != null &&
                               err.message!.isNotEmpty) {
-                            errorMessage = err.details!;
+                            var resp =  await WalletSDKPlugin.parseWalletSDKError(localizedErrorMessage: err.details);
+                            if(resp.category == "INVALID_GRANT"){
```

Comment:
There's one more error category you can add a check for: `INVALID_TOKEN`. In the case of `INVALID_TOKEN`, however, re-entering the PIN won't help, so in that case the error message should just tell the user to scan a new QR code.